### PR TITLE
Implement CRUD for pipelines

### PR DESCRIPTION
**Pull request by haines**
Adds support for all the endpoints under `/pipelines`.

I've added structs for the provider settings but I'm not sure what the API offers for things like GitHub Enterprise. I'd appreciate some help from someone who knows!

I toyed with the idea of just returning `map[string]interface{}` for settings of unknown providers, but it was awkward to work with, so I think it's better just to return `nil`. The client doesn't currently return provider settings so I think this is a step forward in any case.

### CODE REVIEW

**Comment by lox** [comment]
Shall we consolidate on #21?

File: buildkite/pipelines.go [hunk 1]
```diff
@@ -90,3 +98,73 @@ func (ps *PipelinesService) List(org string, opt *PipelineListOptions) ([]Pipeli
...
+		return nil, resp, err
+	}
+
+	return pipeline, resp, err
+}
+
+// Edit a pipeline by slug for a given organization.
+//
+// buildkite API docs: https://buildkite.com/docs/rest-api/pipelines#update-a-pipeline
+func (ps *PipelinesService) Edit(org string, slug string, p *Pipeline) (*Pipeline, *Response, error) {
```
Comment (lox) [inline buildkite/pipelines.go]: I think this should be `Update`, unless there is prior art for `Edit` in the other endpoints.

**Comment by lox** [comment]
Actually, I just dug a bit more into this one. I think I actually like the approach of typed structures for the different providers. Thoughts @mubeta06?

**Comment by lox** [comment]
🤔That said, I think #22 is probably more in-line with the rest of the API and is the better incremental change. I'm going to merge that one and then we can look at ways to bring in some type safety to the `ProviderSettings` map.

**Comment by haines** [comment]
Sweet, I'll raise a new PR with the provider settings.

File: buildkite/pipelines.go (hunk 1 above)
Comment (haines) [reply buildkite/pipelines.go]: I don't think there were any `Update` or `Edit` endpoints anywhere in this client yet, and personally I prefer `Update` too, but I was going off what `go-github` does 🤷‍♂️

**Pull request closed without merging**